src/Elastic.Apm.Extensions.Hosting/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using Elastic.Apm.Api;
+using Elastic.Apm.Config;
+using Elastic.Apm.DiagnosticSource;
+using Elastic.Apm.Extensions.Hosting.Config;
+using Elastic.Apm.Logging;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace Elastic.Apm.Extensions.Hosting
+{
+	public static class HostBuilderExtensions
+	{
+		public static IHostBuilder UseElasticApm(this IHostBuilder builder, params IDiagnosticsSubscriber[] subscribers)
+		{
+			builder.ConfigureServices((ctx, services) =>
+			{
+				services.AddSingleton<IApmLogger, NetCoreLogger>();
+				services.AddSingleton<IConfigurationReader>(sp =>
+					new MicrosoftExtensionsConfig(ctx.Configuration, sp.GetService<IApmLogger>(), ctx.HostingEnvironment.EnvironmentName));
+
+				services.AddSingleton(sp =>
+				{
+					var components = new AgentComponents(sp.GetService<IApmLogger>(), sp.GetService<IConfigurationReader>());
+					UpdateServiceInformation(components.Service);
+					return components;
+				});
+				services.AddSingleton<IApmAgent, ApmAgent>(sp =>
```

Comment:
Should we maybe also add the `ITracer` here?

Reason I bring this up is that I see people a lot being focused on the Agent class itself (or `IApmAgent`), but what they typically want is just to start traces, spans, and capture errors and for that all they need is an `ITracer`.

Reply:
Added